embed/etcd.go

```diff
@@ -82,7 +82,7 @@ type Etcd struct {
 type peerListener struct {
 	net.Listener
 	serve func() error
-	close func(context.Context) error
+	close func(time.Duration) error
```

Comment:
not sure why do we need this change. can you explain?

Reply:
Previous `(e *Etcd) stopGRPCServer(gs *grpc.Server)` method was using `embed.Etcd` only to compute request timeout. Now, we want to use this same function with `*http.Server` parameter. Especially, in line https://github.com/coreos/etcd/pull/8986/files#diff-d7a47eb75475dba263540eb7f8456e50R258.

If we have `time.Duration` as an argument, it's easier to use this function for all shutdown calls.

Reply:
http server shutdown takes a context: https://golang.org/pkg/net/http/#Server.Shutdown

we should do the same, no?